To: vim_dev@googlegroups.com
Subject: Patch 7.3.194
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.194
Problem:    When "b" is a symlink to directory "a", resolve("b/") doesn't
	    result in "a/".  (ZyX)
Solution:   Remove the trailing slash. (Jean-Rene David)
Files:	    src/eval.c


*** ../vim-7.3.193/src/eval.c	2011-05-19 13:40:47.000000000 +0200
--- src/eval.c	2011-05-19 14:55:27.000000000 +0200
***************
*** 14980,14986 ****
--- 14980,14989 ----
  
  	len = STRLEN(p);
  	if (len > 0 && after_pathsep(p, p + len))
+ 	{
  	    has_trailing_pathsep = TRUE;
+ 	    p[len - 1] = NUL; /* the trailing slash breaks readlink() */
+ 	}
  
  	q = getnextcomp(p);
  	if (*q != NUL)
*** ../vim-7.3.193/src/version.c	2011-05-19 14:50:49.000000000 +0200
--- src/version.c	2011-05-19 14:54:40.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     194,
  /**/

-- 
Momento mori, ergo carpe diem

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
